Python/pylifecycle.c

```diff
@@ -1835,8 +1836,16 @@ Py_FinalizeEx(void)
      * Alas, a lot of stuff may still be alive now that will be cleaned
      * up later.
      */
+
+    FILE *fp = stderr;
+    if (dump_file) {
+        char dump_file_name[255];
+        time_t now = time(NULL);
+        sprintf(dump_file_name, "cpython-tracerefs-%ld.dump", now);
+        fp = fopen(dump_file_name,"w");
```

Comment:
I don't see how dump_refs is set to non-zero.

I would prefer to call `_Py_PrintReferences(fp); fclose(fp);` here. So both options are not exclusive.

Reply:
Because we have to call `_Py_PrintReferenceAddresses(fp)` also.
And `python_dump_dir` is optional.